src/main/java/uk/co/lucasweb/aws/v4/signer/Signer.java

```diff
@@ -136,21 +155,46 @@ public Builder headers(Header... headers) {
             return this;
         }
 
-        public Signer build(HttpRequest request, String service, String contentSha256) {
+        public Signer buildQueryString(HttpRequest request, String service, String contentSha256, String date, int expiresSeconds) {
+            CanonicalHeaders canonicalHeaders = getCanonicalHeaders();
+            String dateWithoutTimestamp = formatDateWithoutTimestamp(date);
+            AwsCredentials awsCredentials = getAwsCredentials();
+            CredentialScope scope = new CredentialScope(dateWithoutTimestamp, service, region);
+            String rawQuery = request.getRawQuery();
```

Comment:
I think there might still be a gap here with [sub-delims](https://tools.ietf.org/html/rfc3986#appendix-A). They're allowable per rfc3986 as query param values (so `java.net.URI` will tolerate them), but AWS demands in 3b of https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html that they're percent-encoded.

I ended up decoding then re-encoding them with `URIEncoder.encode(uri.getQuery)`.